### PR TITLE
Add release-1.7 branch config for CAPM3 and IPAM

### PR DESCRIPTION
**Pull request by mquhuy**
/hold

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +188,7 @@ The following table describes which branches are tested for different test trigg
 | test suffix | CAPM3 branch | IPAM branch  | BMO branch/tag  | Keepalived tag | Ironic tag |
 | ----------- | ------------ | -----------  | --------------- | -------------- | ---------- |
 | main        | main         | main         | main            | latest         | latest     |
+| release-1-7 | release-1.7  | release-1.7  | release-0.6     | v0.6.0         | v24.1.1    |
 | release-1-6 | release-1.6  | release-1.6  | release-0.5     | v0.5.1         | v24.1.0    |
```
[the remark this answers is not in the document]
Comment (tuminoid) [reply README.md:192]: Its fixed in Kashif's PR, ignore.

File: lib/common.sh [hunk 2]
```diff
@@ -158,6 +158,9 @@ elif [[ "${CAPM3RELEASEBRANCH}" == "release-1.5" ]]; then
 elif [[ "${CAPM3RELEASEBRANCH}" == "release-1.6" ]]; then
   export CAPM3BRANCH="${CAPM3BRANCH:-release-1.6}"
   export IPAMBRANCH="${IPAMBRANCH:-release-1.6}"
+elif [[ "${CAPM3RELEASEBRANCH}" == "release-1.7" ]]; then
+  export CAPM3BRANCH="${CAPM3BRANCH:-release-1.7}"
+  export IPAMBRANCH="${IPAMBRANCH:-release-1.7}"
```
Comment (kashifest) [inline lib/common.sh:163]: lets remove release-1.4 as well ?

File: lib/common.sh [hunk 3]
```diff
@@ -277,6 +280,13 @@ elif [[ "${CAPM3RELEASEBRANCH}" = "release-1.6" ]]; then
   export KEEPALIVED_TAG="v0.5.1"
   export IRONIC_TAG="v24.0.0"
   export BMOBRANCH="${BMORELEASEBRANCH:-release-0.5}"
+elif [[ "${CAPM3RELEASEBRANCH}" = "release-1.7" ]]; then
```
Comment (kashifest) [inline lib/common.sh:283]: same in this if-else block , lets remove release-1.4 as well

File: lib/releases.sh [hunk 4]
```diff
@@ -81,6 +81,10 @@ elif [ "${CAPM3RELEASEBRANCH}" = "release-1.6" ]; then
   # 1.6.99 points to the head of the release-1.6 branch. Local override for CAPM3 is created for this version.
   export CAPM3RELEASE="v1.6.99"
   export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.6.")}"
+elif [ "${CAPM3RELEASEBRANCH}" = "release-1.7" ]; then
```
Comment (kashifest) [inline lib/releases.sh:84]: same

File: lib/common.sh (hunk 2 above)
Comment (mquhuy) [reply lib/common.sh:163]: Yep

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Review by kashifest — COMMENTED** [review]
/approve

**Comment by tuminoid** [comment]
/hold
for tide conflict, unhold when tests passed

**Comment by tuminoid** [comment]
/hold
tide is not cooperating at all.

**Comment by tuminoid** [comment]
/test metal3-centos-e2e-integration-test-release-1-6 
/unhold
Prow and GH branch protection mismatched.

**Pull request merged**